presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/CanonicalizeExpressionRewriter.java

```diff
@@ -42,6 +45,38 @@ public static Expression canonicalizeExpression(Expression expression)
     private static class Visitor
             extends ExpressionRewriter<Void>
     {
+        @Override
+        public Expression rewriteComparisonExpression(ComparisonExpression node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            // if we have a comparison of the form <constant> <op> <expr>, normalize it to
+            // <expr> <op-flipped> <constant>
+            if (node.getLeft() instanceof Literal && !(node.getRight() instanceof Literal)) {
+                return new ComparisonExpression(
```

Comment:
```suggestion
                node = new ComparisonExpression(
```

otherwise it's not clear whether expressions being flipped are treated the same way as those that did not require flipping